lib/strscan.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+if RUBY_PLATFORM == "java"
+  require 'strscan.jar'
+  JRuby::Util.load_ext("org.jruby.ext.strscan.StringScannerLibrary")
+else
+  require "strscan.so"
+end
+
+class StringScanner
+  # call-seq:
+  #   scan_integer(base: 10)
+  # 
+  # If `base` isn't provided or is `10`, then it is equivalent to calling `#scan` with a `[+-]?\d+` pattern,
+  # and returns an Integer or nil.
+  #
+  # If `base` is `16`, then it is equivalent to calling `#scan` with a `(0x)?[0-9a-fA-F]+` pattern,
```

Comment:
Could you accept sign?
`[+-](0x)?[0-9a-fA-F]+`

It seems sign can be used: https://en.wikipedia.org/wiki/Hexadecimal#Signs

Reply:
Done.